packager/media/chunking/chunking_handler.cc

```diff
@@ -144,13 +147,17 @@ Status ChunkingHandler::OnMediaSample(
   return DispatchMediaSample(kStreamIndex, std::move(sample));
 }
 
-Status ChunkingHandler::EndSegmentIfStarted() const {
+Status ChunkingHandler::EndSegmentIfStarted(bool fromCueEvent) const {
   if (!segment_start_time_)
     return Status::OK;
 
   auto segment_info = std::make_shared<SegmentInfo>();
   segment_info->start_timestamp = segment_start_time_.value();
   segment_info->duration = max_segment_time_ - segment_start_time_.value();
+  // If fromCueEvent then current_segment_index_ is already added to the
+  // number of segments before last cue.
+  segment_info->segment_index = ((fromCueEvent) ? 0 : current_segment_index_) +
+                                num_segments_before_last_cue_ - 1;
```

Comment:
I am subtracting 1 to send the segment index as 0 based. To remove -1, I can move the lines 
`   |108       current_segment_index_ = segment_index; `                                                    
`    |109       // Reset subsegment index.                          `                                          
`    |110       current_subsegment_index_ = 0; ` 
in ChunkingHandler::OnMediaSample after the call to  
   `|112 RETURN_IF_ERROR(EndSegmentIfStarted());`

Reply:
Yes, I think the change you mentioned is needed. `current_segment_index_` should also be set to -1 instead of 0 in OnCueEvent.

Here is a simple illustration with segment size of 2. 

(| is segment boundary; C is for cue).

```
_ _ | _ C _ _ | _ _
1   | 2  |  3 | 4
```

Region 1: current_segment_index_ = 0, num_segments_before_last_cue_ = 0 so segment_info->segment_index = 0
Region 2: current_segment_index_ = 1 => num_segments_before_last_cue_ = 2, current_segment_index_=-1 so segment_info->segment_index = 1
Region 3: current_segment_index_ = 0, num_segments_before_last_cue_ = 2 so segment_info->segment_index = 2
Region 4: current_segment_index_ = 1, num_segments_before_last_cue_ = 2 so segment_info->segment_index = 3

If we don't have unit-tests, we should add unit-tests for it.

Reply:
@kqyang Thanks for the illustration. I have made this change. 
Also looks like there is a unit test for this: ChunkingHandlerTest,CueEvent.